internal/controller/stas/scan_job_controller.go

```diff
@@ -53,6 +58,81 @@ func (r *ScanJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				ignoreDeletionPredicate(),
 			)).
 		Complete(r.reconcile())
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
wonder if we should move this controller logic into its own file called `cleanup_jobs.go`. Keeps the main logic simple and easy to follow?

Reply:
I actually want to refactor the whole thing afterwards. For now we need to use the same Reconciler struct to be able to reuse code.

Reply:
sounds good